### DISCUSSION TITLE
how to set the asset_root(default or nvidia) to a local path in isaacSim 4.0.0?

### DISCUSSION THREAD

**Discussion by xibeisiber** [isaac-sim/IsaacLab#534] [Q&A] [answered]
Hi, I installed  Isaac Sim using the PIP method following instruction in https://isaac-sim.github.io/IsaacLab/source/setup/installation/pip_installation.html#installing-isaac-sim. 

When trying to run the example "./isaaclab.sh -p source/standalone/demos/arms.py", the error always occurs:
```
2024-06-16 13:24:33 [260,341ms] [Warning] [omni.usd] Warning: in _ReportErrors at line 2890 of /buildAgent/work/ac88d7d902b57417/
USD/pxr/usd/usd/stage.cpp -- In </World/defaultGroundPlane>: Could not open asset @http://omniverse-content-production.s3-us-
west-2.amazonaws.com/Assets/Isaac/4.0/Isaac/Environments/Grid/default_environment.usd@ for reference introduced by 
@anon:0x14f98080:World0.usd@</World/defaultGroundPlane>. (recomposing stage on stage @anon:0x14f98080:World0.usd@ 
<0x1508bf00>)

Traceback (most recent call last):
  File "/home/jia/Workspace/reinforcement_learning/IsaacLab/source/standalone/demos/arms.py", line 230, in <module>
    main()
  File "/home/jia/Workspace/reinforcement_learning/IsaacLab/source/standalone/demos/arms.py", line 218, in main
    scene_entities, scene_origins = design_scene()
  File "/home/jia/Workspace/reinforcement_learning/IsaacLab/source/standalone/demos/arms.py", line 79, in design_scene
    cfg.func("/World/defaultGroundPlane", cfg)
  File "/home/jia/Workspace/reinforcement_learning/IsaacLab/source/extensions/omni.isaac.lab/omni/isaac/lab/sim/spawners/from_files/from_files.py", line 145, in spawn_ground_plane
    collision_prim_path = prim_utils.get_prim_path(
  File "/home/jia/miniforge3/envs/isaaclab/lib/python3.10/site-packages/isaacsim/exts/omni.isaac.core/omni/isaac/core/utils/prims.py", line 631, in get_prim_path
    return prim.GetPath()
AttributeError: 'NoneType' object has no attribute 'GetPath'
```

Therefore I download the isaac-sim-assets-4.0.0(1-4) and merge into one local folder and intend to let Isaac sim find USD files from this local folder instead of "http://omniverse-content-production.s3-us-west-2.amazonaws.com".

I modified in `envs/isaaclab/lib/python3.10/site-packages/omni/data/Kit/Isaac-Sim/4.0/user.config.json` the `persistent.isaac.asset_root.default` and `persistent.isaac.asset_root.nvidia` to my local path. But it seems everytime after I run `"./isaaclab.sh -p source/standalone/demos/arms.py"`, the `user.config.json` file will be overwriten (twice) with "default": "http://omniverse-content-production.s3-us-west-2.amazonaws.com/Assets/Isaac/4.0" again, so it never reads USD files from my local path..

**Comment by SYMEngrave** [comment]
I installed the lab using the binaries way, but I encountered the same issue when running arms.py. This is very frustrating, and I am currently at a loss.

**Reply by caozx1110** [reply]
Same problem

**Comment by pxlong** [comment, marked as answer]
@SYMEngrave @xibeisiber @caozx1110 
Hi, I figure out a solution for this issue:
1. download all assets packs to your local computer: https://docs.omniverse.nvidia.com/isaacsim/latest/installation/install_faq.html#assets-pack
2. change the **NUCLEUS_ASSET_ROOT_DIR** in `IsaacLab/source/extensions/omni.isaac.lab/omni/isaac/lab/utils/assets.py` to _the_path_to_your_assets_packs_, then IsaacLab will use your local assets instead of the assets from AWS:  
`NUCLEUS_ASSET_ROOT_DIR = "/home/xxx/App/isaac/isaac-sim-assets-1-4.0.0/Assets/Isaac/4.0"`

**Reply by xibeisiber** [reply]
thanks! it works for me

**Reply by pxlong** [reply]
glad to hear that! you're welcome
